crates/adapters/src/controller/mod.rs

```diff
@@ -286,6 +286,22 @@ impl Controller {
         self.inner.num_api_connections()
     }
 
+    /// Force the circuit to perform a step even if all of its
+    /// input buffers are empty or nearly empty.
+    ///
+    /// Can be used in two situations:
+    ///
+    /// * To process scalar inputs, such as neighborhood or quantile querues that
+    ///   are not reflecred in input buffer sizes.
+    ///
+    /// * When input buffers are not completely empty, but contain fewere than
+    ///   `min_batch_size_records` records the circuit will normally wait for a small
+    ///   timeout before processing the data.  This function can be used to force
+    ///   immediate processing.
+    pub fn request_step(&self) {
```

Comment:
is this really different from step?

Reply:
I guess it does some additional preparatory work

Reply:
`step` happens in the context of a separate thread.  This API can be invoked from any thread to wake up that thread and force it to call `step`.